src/core/src/main/java/org/apache/jmeter/testelement/AbstractTestElement.java

```diff
@@ -342,7 +343,7 @@ public int getPropertyAsInt(String key) {
 
     @Override
     public int getPropertyAsInt(String key, int defaultValue) {
-        JMeterProperty jmp = getRawProperty(key);
+        JMeterProperty jmp = getPropertyOrNull(key);
         return jmp == null || jmp instanceof NullProperty ? defaultValue : jmp.getIntValue();
```

Comment:
Frankly speaking, `getPropertyOrNull` followed by `instanceof NullProperty` look strange, however, I do not change it in the current PR.
However, it would probably make sense to revise it later.